test/test_query2.cpp

```diff
@@ -6056,44 +6056,93 @@ TEST(Query_FullTextMulti)
+        std::vector<int64_t> keys(tv.size());
+        for (size_t i = 0; i < tv.size(); ++i)
+            keys[i] = tv.get_key(i).value;
+        return keys;
+    };
+    auto do_fulltext_find = [&](StringData term) -> Keys {
+        return get_keys(table->where().fulltext(col, term).find_all());
+    };
+    auto do_query_find = [&](const TableRef& table, StringData query) -> Keys {
+        return get_keys(table->query(query).find_all());
+    };
+
+    CHECK_THROW_ANY(do_fulltext_find(""));
+
     // search with multiple terms
-    auto tv = table->where().fulltext(col, "object gemstone").find_all();
-    CHECK_EQUAL(2, tv.size());
+    CHECK_EQUAL(do_fulltext_find("one three"), Keys({7, 8}));
+    CHECK_EQUAL(do_fulltext_find("three one"), Keys({7, 8}));
+    CHECK_EQUAL(do_fulltext_find("1990s"), Keys({2, 4}));
+    CHECK_EQUAL(do_fulltext_find("1990s c++"), Keys({4}));
+    CHECK_EQUAL(do_fulltext_find("object gemstone"), Keys({2, 4}));
 
     // over links
-    tv = origin->link(col_link).column<String>(col).fulltext("object gemstone").find_all();
-    CHECK_EQUAL(2, tv.size());
-
-    tv = origin->query("link.text TEXT 'object gemstone'").find_all();
-    CHECK_EQUAL(2, tv.size());
+    CHECK_EQUAL(do_query_find(origin, "link.text TEXT 'object gemstone'"), Keys({2, 4}));
+    auto tv = origin->link(col_link).column<String>(col).fulltext("object gemstone").find_all();
+    CHECK_EQUAL(get_keys(tv), Keys({2, 4}));
 
     // through LnkLst
     auto obj = tv.get_object(0);
     auto ll = obj.get_linklist(col_link);
     tv = table->where(ll).fulltext(col, "object gemstone").find_all();
-    CHECK_EQUAL(1, tv.size());
+    CHECK_EQUAL(get_keys(tv), Keys({2}));
 
     // Diacritics ignorant
-    tv = table->where().fulltext(col, "depot emanant").find_all();
-    CHECK_EQUAL(1, tv.size());
+    CHECK_EQUAL(do_fulltext_find("depot emanant archive"), Keys({5}));
 
     // search for combination that is not present
-    tv = table->where().fulltext(col, "object data").find_all();
-    CHECK_EQUAL(0, tv.size());
+    CHECK_EQUAL(do_fulltext_find("object data"), Keys());
+
+    // exclude words
+    CHECK_EQUAL(do_fulltext_find("-three one"), Keys({9}));
+    CHECK_EQUAL(do_fulltext_find("one -three"), Keys({9}));
+    CHECK_EQUAL(do_fulltext_find("object -databases"), Keys({1, 2, 4, 6}));
+    CHECK_EQUAL(do_fulltext_find("-databases object -duplicates"), Keys({1, 2, 4}));
+    CHECK_EQUAL(do_fulltext_find("object -objects"), Keys({2, 4, 6}));
+    CHECK_EQUAL(do_fulltext_find("-object objects"), Keys({3}));
+    CHECK_EQUAL(do_fulltext_find("databases -database"), Keys({}));
+    CHECK_EQUAL(do_fulltext_find("-database databases"), Keys({}));
+    CHECK_EQUAL(do_fulltext_find("database -databases"), Keys({1, 2, 4}));
+    CHECK_EQUAL(do_fulltext_find("-databases database"), Keys({1, 2, 4}));
+
+    // invalid exclude searches
+    // There should always be at least one mandatory term
+    CHECK_THROW_ANY(do_fulltext_find("-databases"));
```

Comment:
So with this @jedelbo mentioned in linked pr that: 

"At some point we agreed that there must be at least one mandatory search word. I can see that that decision did not make it into the scope doc."

I think it can be argued that one mandatory search work is there, and it's 'databases'. The addition of '-' is simply an instruction for search to invert match and return all keys, which don't contain 'databases' in them. To me it looks like pretty natural thing to expect from exclusion of terms. But apparently there is quite a room for interpretation from original design of this feature.

@nirinchev what do you think about this behavior? Should it throw and doesn't allow such search?

Reply:
I think we should prevent this. If a user by mistake adds the minus in front of a token, the query could be expensive. If it is used in a FLX request, on a big database you will get potentially get very many objects downloaded. Imagine you are searching a store for all items not including "umbrella" in the description.

But if you want this behavior, you can get it.

Reply:
Just for clarification, would the correct way of doing this be "* -umbrella" ?

Reply:
Right now we don't support FTS in sync queries, so I wouldn't worry too much about it. Even then, we don't prevent you from writing a query like `realm.all<Person>("age != 0")`, which would similarly download a lot of records. So if we can support it without a whole lot of effort, I'd vote to do so.

Reply:
You got it!